commands/http/parse.go

```diff
@@ -20,12 +20,20 @@ func Parse(r *http.Request, root *cmds.Command) (cmds.Request, error) {
 
 	stringArgs := make([]string, 0)
 
+	if err := apiVersionMatches(r); err != nil {
```

Comment:
> I would argue that an api version check belongs in the commands lib. Thats a nice feature for the commands library to provide. Being able to have your automatically generated http api do version sanity checks is pretty slick.

agreed there, but then this should be properly done so it does not include https://github.com/ipfs/go-ipfs/pull/2037/files#diff-4df02c629d914567df45314a464367b9R17